src/drivers/ws2812.rs

```diff
@@ -30,12 +47,14 @@ impl<SPI: SpiBus<u8>> Ws2812<SPI> {
     }
 }
 
-impl<SPI: SpiBus<u8>> LedDriver<RGB8> for Ws2812<SPI> {
+impl<'a, SPI: SpiBus<u8>> LedDriver<RGB8> for Ws2812<'a, SPI> {
     type Error = SPI::Error;
 
     async fn write(&mut self, colors: &[RGB8]) -> Result<(), Self::Error> {
-        let num_leds = core::cmp::min(colors.len(), MAX_LEDS);
+        let num_leds = core::cmp::min(colors.len(), self.num_leds);
         let data_size = num_leds * 12;
+        let reset_size = 140;
```

Comment:
is the reset_size fixed or is it dependent on the amount of leds somehow?

Reply:
> is the reset_size fixed or is it dependent on the amount of leds somehow?

It's a fixed value, it does not depend on the led number. Basically it represent the amount of time to recognize the end of a data transsmission and reset for the next frame